code/datums/gamemode/misc_gamemode_procs.dm

```diff
@@ -361,7 +361,7 @@
 			traitor_mob.mind.total_TC += R.hidden_uplink.uses
 		if (role && T)
 			role.uplink = T
-
+		T.job = role.name
```

Comment:
indent 1 time

Reply:
while I do like your idea of reducing the number of vars where possible, you can't eliminate this one; the old var you just killed made challengers behave differently from other traitors. In other words, this is what would happen.

challenger -> name_for_uplink isn't null -> uplink gets that name instead of a job name
normal traitor -> name_for_uplink is null -> uplink assigns real job

the new system does not differentiate, meaning all uplinks for normal tots will just be treated as having job "traitor", you've effectively eliminated all job bonuses for normal tots

Reply:
done, compiled, and tested